### ISSUE TITLE
DNS leaks for every request to ISP DNS - Produced by Adguard Home v0.107.3

### ISSUE THREAD

**Issue by eggplanting** [AdguardTeam/AdGuardHome#4339]
### Issue Details

* **Version of AdGuard Home server:**
  * 0.107.3
* **How did you install AdGuard Home:**
  * Docker
* **How did you setup DNS configuration:**
  * Adguard Home as DHCP server

### Expected Behavior
Adguard Home will only use the Upstream DNS servers configured manually in DNS settings

### Actual Behavior
Adguard Home is using the configured Upstream DNS and also uses the DNS servers of my ISP, which are hardcoded to my router. When manually setting the DNS server to Adguard Home it still is leaking all DNS requests, so the issue must be produced by Adguard Home. It makes no difference if the client receives the DNS via Adguard DHCP or if it Adguard DHCP is entered manually. This is not happening with a Pi-hole setup.

### Screenshots

![1-AdguardSettings](https://user-images.githubusercontent.com/57031979/155865579-d1c28983-58ec-4a4d-9f25-c36bcad65369.PNG)

![2-DNSleak](https://user-images.githubusercontent.com/57031979/155865584-dcdee188-e05c-4776-a3f0-2fcf9a434b77.PNG)

![3-DNSleak](https://user-images.githubusercontent.com/57031979/155865588-38d32713-8315-4590-a1cd-e8596433fe2f.PNG)

![4-DHCP](https://user-images.githubusercontent.com/57031979/155865589-1df6ab39-d161-4141-b642-e0d9f8cb1088.PNG)

### Additional Information
This is an heavy privacy issue which needs to be fixed ASAP

**Comment by agneevX** [comment]
In your upstream configuration, there's a single DoT server, yet you also say this:

> Adguard Home is using the configured Upstream DNS and also uses the DNS servers of my ISP, which are hardcoded to my router.

So which is it? It seems to me that AdGuard does not receive your requests at all.

Did you check Query Log?

**Comment by eggplanting** [comment]
Hi @agneevX 

Adguard Home receives every request of my client, I can see them in the Query Log.
Running a DNS Leak Test from my client still results with the same, the upstream DNS of Adguard and my IPS provider DNS is shown. Like in the screenshot above.

I configured Adguard Home as my only DNS server manually on my client network interfaces and see the queries for each DNS request on Adguard, but it still seems to fallback or request the ISP DNS parallely.

What I mean with hardcoded DNS on the router is, that I cannot disable the DNS server on my ISP router.
So Adguard needs to be able to not forward the DNS requests to my Gateway, I don't have the mentioned issue with Pi-hole.

**Comment by agneevX** [comment]
AdGuard Home does not forward queries anywhere but to upstreams.

I think your router is intercepting plaintext DNS requests that have their destination as port 53 tcp/udp, and sending them to multiple destinations including AdGuard Home and perhaps your ISP's own hardcoded DNS, as you said. 
While doing this, it perhaps only return the fastest IP received. DoH/DoT is slower than traditional DNS.

**Comment by eggplanting** [comment]
It really seems that Adguard is not the issue, after testing the same setup with Pi-hole it's also happening there.

My provider router seems to reject any DNS 53 traffic and uses it's own DNS server instead:
https://forum.vodafone.de/t5/Internet-Ger%C3%A4te/Vodafone-Station-blockiert-lokalen-DNS-Resolver-Pi-Hole/td-p/2826339

That should be a legal issue in my eyes, I never have seen something like that.
You can't even disable this feature on my router.

**Comment by agneevX** [comment]
Yeah that's very bad indeed. Can you disable DHCP on the router and use AdGuard's DHCP instead?

**Comment by eggplanting** [comment]
That's what I did, DHCP addresses are and were already answered only by Adguard. I disable DHCP on my router and blocked it via DHCP snooping on my switch.

What happens is that DNS requests are really intercepted by my router, this also happens when I assign a static IP of an Upstream DNS server on a local client. Only my Vodafone DNS responds, that's crazy.

**Comment by agneevX** [comment]
You should get a router instead of using your ISP's modem tbh. Who knows what sort of software they have installed on it that uploads personal data to their servers.

**Comment by eggplanting** [comment]
Found the root cause.

The issue is happening only for IPv6 enabled clients, my Vodafone router assigns a IPv6 DNS server which is requested instead my local Adguard IPv4 server.

Wow this is bad, I need to exchange this router.
